src/cmem/ActivityControl/ActivityControlWidget.tsx

```diff
@@ -102,21 +103,16 @@ export function ActivityControlWidget(props: IActivityControlProps) {
         border,
         canShrink,
         tags,
+        progressFinishedIcon,
     } = props;
-
     const spinnerClassNames = (progressSpinner?.className ?? "") + ` ${eccgui}-spinner--permanent`;
     const widget = (
         <OverviewItem data-test-id={dataTestId} hasSpacing={border} densityHigh={small}>
             {progressBar && <ProgressBar {...progressBar} />}
             {progressSpinner && (
                 <OverviewItemDepiction keepColors>
-                    {progressSpinner?.value ? (
-                        <Icon
-                            name={`state-${progressSpinner.intent}`}
-                            intent={progressSpinner.intent as IntentTypes}
-                            large={!small}
-                            small={small}
-                        />
+                    {progressFinishedIcon && progressSpinner?.value === 1 ? (
```

Comment:
The condition `progressSpinner?.value === 1` is  probably not necessary. It only leads to the situation that a spinner is necessary even if the widget uses a progress bar. I would suggest not to check for spinner value at all. If there is an icon set then replace the spinner. From outside the element only should be updated with an icon when the spinner should not be shown.